service/history/workflowExecutionContext.go

```diff
@@ -927,3 +932,35 @@ func (c *workflowExecutionContextImpl) emitSessionUpdateStats(stats *persistence
 	c.metricsClient.RecordTimer(metrics.SessionCountStatsScope, metrics.DeleteRequestCancelInfoCount,
 		time.Duration(stats.DeleteRequestCancelInfoCount))
 }
+
+func (c *workflowExecutionContextImpl) validateNoEventsAfterWorkflowFinish(input []*workflow.HistoryEvent) error {
+	if len(input) == 0 {
+		return nil
+	}
+
+	seenWorkflowCompletionEvent := false
+	for _, event := range input {
+		if seenWorkflowCompletionEvent {
+			c.logger.WithFields(bark.Fields{
+				logging.TagDomainID:            c.domainID,
+				logging.TagWorkflowExecutionID: c.workflowExecution.GetWorkflowId(),
+				logging.TagWorkflowRunID:       c.workflowExecution.GetRunId(),
+			}).Warn("encount case where events appears after workflow finish.")
+
+			return &shared.InternalServiceError{Message: "error validating last event being workflow finish event."}
```

Comment:
Can you define an error in the beginning of file.  Let's use the same pattern we follow in other parts of the code.